cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java

```diff
@@ -0,0 +1,241 @@
+                            ArtifactRepository artifactRepository, ArtifactStore artifactStore,
+                            AuthorizerInstantiator authorizerInstantiator, StreamAdmin streamAdmin,
+                            StreamCoordinatorClient streamCoordinatorClient, PrivilegesManager privilegesManager,
+                            AuthorizationEnforcer authorizationEnforcer) {
+
+    this.cConf = cConf;
+    this.hConf = hConf;
+    this.datasetFramework = datasetFramework;
+    this.discoveryService = discoveryService;
+    this.preferencesStore = preferencesStore;
+    this.secureStore = secureStore;
+    this.transactionManager = transactionManager;
+    this.artifactRepository = artifactRepository;
+    this.artifactStore = artifactStore;
+    this.authorizerInstantiator = authorizerInstantiator;
+    this.streamAdmin = streamAdmin;
+    this.streamCoordinatorClient = streamCoordinatorClient;
+    this.privilegesManager = privilegesManager;
+    this.authorizationEnforcer = authorizationEnforcer;
+  }
+
+  @POST
+  @Path("/previews")
+  public void start(HttpRequest request, HttpResponder responder,
+                    @PathParam("namespace-id") String namespaceId) throws Exception {
+
+    // Create new injector using method createPreviewInjector(applicationId, datasetNames).
+    // Get the instance of the PreviewManager and call start() method on it.
+    // The returned injector will be cached
+  }
+
+  @POST
+  @Path("/previews/{preview-id}/stop")
+  public void stop(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespaceId,
+                   @PathParam("preview-id") String previewId) throws Exception {
+
+  }
+
+  @GET
+  @Path("/previews/{preview-id}/status")
+  public void getStatus(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespaceId,
+                        @PathParam("preview-id") String previewId)  throws Exception {
+
+  }
+
+  @GET
+  @Path("/previews/{preview-id}/loggers")
+  public void getLoggers(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespaceId,
+                         @PathParam("preview-id") String previewId) throws Exception {
+
+  }
+
+  @GET
+  @Path("/previews/{preview-id}/loggers/{logger-id}")
+  public void getData(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespaceId,
+                      @PathParam("preview-id") String previewId, @PathParam("logger-id") String loggerId)
+    throws Exception {
+
+  }
+
+
+  /**
+   * Create injector for the given application id.
+   */
+  @VisibleForTesting
+  public Injector createPreviewInjector(ApplicationId applicationId, Set<String> datasetNames) {
+    CConfiguration previewcConf = CConfiguration.copy(cConf);
+    java.nio.file.Path previewDirPath = Paths.get(cConf.get(Constants.CFG_LOCAL_DATA_DIR), "preview").toAbsolutePath();
```

Comment:
Since it is in `PreviewHttpHandler`, Path clashes with the `javax.ws.rs.Path`. Should I move the `createInjector` method to different class in the same package?